packages/react-ai/src/hooks/useAIConversation.tsx

```diff
@@ -212,43 +225,25 @@ export function createUseAIConversation<
           }
 
           // no ref means its the first event for the message stream
+          // so lets create the contentBlocks ref or else we will
+          // add the incoming event to the right content content block
           if (!contentBlocksRef.current) {
             contentBlocksRef.current = [[event]];
-
-            setDataState((prev) => {
```

Comment:
Combined 2 setDataStates so we only call it once whether or not its the first event